paper/paper.Rmd

```diff
@@ -0,0 +1,171 @@
+    affiliation: 4
+affiliations:
+  - index: 1
+    name: Netflix
+  - index: 2
+    name: Google
+  - index: 3
+    name: Mathematisches Institut der Heinrich-Heine-Universität Düsseldorf
+  - index: 4
+    name: Center for Humans and Machines, Max Planck Institute for Human Development, Berlin, Germany
+output: 
+    md_document:
+      variant: "markdown"
+      preserve_yaml: true
+      standalone: true
+bibliography: paper.bib
+csl: apa.csl
+link-citations: yes
+---
+
+```{r, warning=FALSE, message=FALSE, echo=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  out.width = "100%",
+  comment = "#>"
+)
+
+library(lintr)
+
+withr::local_options(list(
+  lintr.format_width = 60L
+))
+```
+
+# Statement of Need
+
+R is an interpreted, dynamically typed programming language [@base2023]. It is a popular choice for statistical analysis and visualization, and is used by a wide range of researchers and data scientists. The `{lintr}` package is an open-source R package that provides static code analysis to check for a variety of common problems related to readability, efficiency, consistency, style, etc. In particular, it enforces the [tidyverse style guide]((https://style.tidyverse.org/index.html)). It is designed to be easy to use and integrate into existing workflows, and can be run from the command line or used as part of an automated build or continuous integration process. `{lintr}` also integrates with a number of popular IDEs and text editors, such as RStudio and Visual Studio Code, making it convenient for users to run `{lintr}` checks on their code as they work.
+
+# Features
+
+As of this writing, `{lintr}` offers `r length(all_linters())` linters.
+
+```{r}
+library(lintr)
+
+length(all_linters())
+```
+
+Naturally, we can't discuss all of them here. To see details about all available linters, we encourage readers to see <https://lintr.r-lib.org/dev/reference/index.html#individual-linters>.
+
+We will showcase one linter for each kind of common problem found in R code.
+
+- **Best practices**
+
+`{lintr}` offers linters that can detect problematic antipatterns and suggest alternative patterns that follow best practices.
+
+For example, expressions like `ifelse(x, TRUE, FALSE)` and `ifelse(x, FALSE, TRUE)` are redundant; just `x` or `!x` suffice in R code where logical vectors are a core data structure. The `redundant_ifelse_linter()` linter detects such problematic usages.
+
+```{r}
+lint(
+  text = "ifelse(x >= 2.5, TRUE, FALSE)",
+  linters = redundant_ifelse_linter()
+)
+```
+
+```{r, eval=FALSE, echo=FALSE}
+lint(
+  text = "ifelse(x >= 2.5)",
+  linters = redundant_ifelse_linter()
+)
+```
+
+- **Efficiency**
+
+Sometimes the users might not be aware of a more efficient way offered by R for carrying out a computation. `{lintr}` offers linters to provide suggestions to improve code efficiency.
+
+```{r}
+lint(
+  text = "any(is.na(x), na.rm = TRUE)",
+  linters = any_is_na_linter()
+)
+```
+
+```{r, echo=FALSE, eval=FALSE}
```

Comment:
Here, the "no lint" example does add value, but maybe better served by explaining in text why `any(is.na())` is deemed inefficient.